### PR TITLE
ArrayWrapper Constructor, add, get, and set methods + Unit Testing

### PR DESCRIPTION
**Pull request by de02E**

### CODE REVIEW

**Review by pswish — APPROVED** [review]
Nice Work, looks great to me, ship it!

File: dsa-23au/java-dsa/arrays-links/src/test/java/dev/codewithfriends/ArrayWrapperTest.java [hunk 1]
```diff
@@ -9,15 +9,50 @@
...
+
     @Test
-    public void insertShifts()
-    {
-        List l = new ArrayWrapper<Book>(10);
-        assertEquals(0, l.size());
-        l.add(b1);
-        assertEquals(1, l.size());
+    public void testSize(){
+        assertEquals(null, 0, 0, 0);
```
Comment (learner-long-life) [inline dsa-23au/java-dsa/arrays-links/src/test/java/dev/codewithfriends/ArrayWrapperTest.java:44]: For this method about testing the size, try to come up with an `assertEquals` statement that depends on `this.currentSize` and asserts that a newly created list has size 0, and after some inserts and deletes, has the appropriate new size.

This PR is already merged, so I'll open a new issue to describe this work.

File: dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/ArrayWrapper.java [hunk 2]
```diff
@@ -66,17 +78,31 @@ public void clear() {
 
     @Override
     public Object get(int index) {
-        return null;
+        return this._array[index];
```
Comment (learner-long-life) [inline dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/ArrayWrapper.java:81]: 👍

File: dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/ArrayWrapper.java [hunk 3]
```diff
@@ -66,17 +78,31 @@ public void clear() {
...
     @Override
     public Object set(int index, Object element) {
-        return null;
+        if (index > maxSize) {
+            return null;
+        }
+        else {
+            Object previousElement = this._array[index];
+            this._array[index] = element;
+            return previousElement;
```
Comment (learner-long-life) [inline dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/ArrayWrapper.java:92]: Nice, choosing to return the replaced element makes a lot of sense, as opposed to returning the new element.

File: dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/ArrayWrapper.java [hunk 4]
```diff
@@ -66,17 +78,31 @@ public void clear() {
...
+        }
     }
 
     @Override
     public void add(int index, Object element) {
-
+        if (this.currentSize != this.maxSize) {
+            for (int i=0; i < this.currentSize-(index);i++){
+                _array[this.currentSize-i] = _array[this.currentSize-i-1];
```
Comment (learner-long-life) [inline dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/ArrayWrapper.java:100]: Nice work! It's safer to check `this.currentSize < this.maxSize` but your way definitely works.

**Pull request merged**